service/githubhook/githubhook.go

```diff
@@ -0,0 +1,100 @@
+// Package githubhook implements handling and verification of github webhooks.
+package githubhook
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	signaturePrefix = "sha1="
+	signatureLength = 45 // len(SignaturePrefix) + len(hex(sha1))
+)
+
+// Hook is an inbound github webhook.
+type Hook struct {
```

Comment:
We usually name files after the struct name. `hook.go` in this case. The same applies to `Oncall` type.